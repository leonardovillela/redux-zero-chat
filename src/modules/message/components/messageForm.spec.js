import React from 'react';
import { shallow } from 'enzyme';
import Chance from 'chance';

import MessageForm from './messageForm';

describe('Test MessageForm component unit', () => {
  const chance = new Chance();

  it('Should render send button disabled when messageContent is empty', () => {
    const wrapper = shallow(<MessageForm messageContent="" />);

    expect(wrapper.find('button').prop('disabled')).toBeTruthy();
  });

  it('Should render send button disabled when messageContent is not empty', () => {
    const messageStub = chance.string();
    const wrapper = shallow(<MessageForm messageContent={messageStub} />);

    expect(wrapper.find('button').prop('disabled')).toBeFalsy();
  });

  it('Should invoke setMessageContent function when onChange is trigged', () => {
    const eventStub = Object.freeze({ target: { value: chance.string() } });
    const setMessageContentMock = jest.fn();
    const wrapper = shallow(<MessageForm setMessageContent={setMessageContentMock} />);

    wrapper.find('textarea').simulate('change', eventStub);
    expect(setMessageContentMock).toBeCalledWith(eventStub.target.value);
  });

  it('Should invoke createMessage function when form is submitted', () => {
    const eventPreventDefaultMock = jest.fn();
    const createMessageMock = jest.fn();
    const eventStub = Object.freeze({ preventDefault: eventPreventDefaultMock });
    const wrapper = shallow(<MessageForm createMessage={createMessageMock} />);

    wrapper.find('form').simulate('submit', eventStub);
    expect(eventPreventDefaultMock).toBeCalled();
    expect(createMessageMock).toBeCalled();
  });

  it('Should render component with correct structure', () => {
    const wrapper = shallow(
      <MessageForm
        className="Teste"
        messageContent="Msg"
        clearMessageContent={jest.fn()}
        setMessageContent={jest.fn()}
        createMessage={jest.fn()}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
