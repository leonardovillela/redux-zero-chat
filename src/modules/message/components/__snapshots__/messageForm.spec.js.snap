// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MessageForm component unit Should render component with correct structure 1`] = `
<div
  className="Teste"
>
  <form
    className="h-100"
    onSubmit={[Function]}
  >
    <textarea
      className="w-80 h-100 v-top pa3"
      onChange={[Function]}
      value="Msg"
    />
    <button
      className="w-20 h-100 v-top"
      disabled={false}
    >
      Enviar
    </button>
  </form>
</div>
`;
